src/framework/core/js/DataBinding.js

```diff
@@ -1507,4 +1507,76 @@ var fluid_3_0_0 = fluid_3_0_0 || {};
         return that;
     };
 
+    /**
+     * Calculates the changes between the model values 'value' and
+     * 'oldValue'. The optional argument 'changePathPrefix' is prepended
+     * to the generated change paths. This is useful for generating
+     * change records to be applied at a non-root path in a model.
+     * Returns an array of change records, suitable for passing to
+     * fluid.fireChanges().
+     */
+    fluid.modelPairToChanges = function (value, oldValue, changePathPrefix) {
+        changePathPrefix = changePathPrefix || "";
+
+        var diffOptions = {changes: 0, unchanged: 0, changeMap: {}};
+        fluid.model.diff(oldValue, value, diffOptions);
+
+        var changes = [];
+
+        fluid.modelPairToChanges1(
+            value,
+            fluid.pathUtil.parseEL(changePathPrefix),
+            diffOptions.changeMap,
+            [],
+            changes
+        );
+
+        return changes;
+    };
+
+    fluid.modelPairToChanges1 = function (value, changePathPrefixSegs, changeMap, changeSegs, changes) {
```

Comment:
The suffix "1" is a little cryptic - perhaps something like modelPathToChangesRecurse or modelPathToChangesImpl? Doc comment also desirable given the long list and complex workflow